src/main/java/TaskManager.java

```diff
@@ -0,0 +1,48 @@
+public class TaskManager {
+    private final Task[] taskList;
+    private int currentTaskIndex;
+
+    public TaskManager(){
+        currentTaskIndex = 0;
+        taskList = new Task[100];
+    }
+
+    private void addTask(Task newTask){
+        taskList[currentTaskIndex++] = newTask;
+        System.out.printf("Got it. I've added this task:%n  %s%nNow you have %d tasks in the list.%n", newTask, currentTaskIndex);
+    }
+
+    public void addTodo(String description){
+        Todo newTodo = new Todo(description);
+        addTask(newTodo);
+    }
+
+    public void addDeadline(String description, String deadlineDate){
+        Deadline newDeadline = new Deadline(description, deadlineDate);
+        addTask(newDeadline);
+    }
+
+    public void addEvent(String description, String eventStart, String eventEnd){
+        Event newEvent = new Event(description, eventStart, eventEnd);
+        addTask(newEvent);
+    }
+
+    public void markTask(int index){
+        Task selectedTask = taskList[index];
+        selectedTask.markDone();
+        System.out.printf("Nice! I've marked this task as done:%n  %s%n", selectedTask);
+    }
+
+    public void unmarkTask(int index){
+        Task selectedTask = taskList[index];
+        selectedTask.unmarkDone();
+        System.out.printf("OK, I've marked this task as not done yet:%n %s%n", selectedTask);
+    }
```

Comment:
The markTask and unmarkTask methods are very similar, with only a slight difference. You could merge them into one method by passing a boolean value for the mark status to produce the required outputs

Reply:
Adding on to the above, maybe reuse that boolean value (for the mark status) to print the 2 different strings depending on whether you are marking or unmarking.